app/src/main/java/com/example/barterbuddy/activities/TradeUnreadPage.java

```diff
@@ -0,0 +1,71 @@
+package com.example.barterbuddy.activities;
+
+import static com.example.barterbuddy.network.UpdateTradeDocument.setStateToCanceled;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
+import com.example.barterbuddy.R;
+import com.example.barterbuddy.models.Trade;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class TradeUnreadPage extends AppCompatActivity {
+  private final FirebaseAuth AUTHENTICATION_INSTANCE = FirebaseAuth.getInstance();
+  private final FirebaseFirestore FIRESTORE_INSTANCE = FirebaseFirestore.getInstance();
+  private FirebaseUser currentUser;
+  private String username;
+  private String email;
+  private Button cancel_button;
+  private Trade trade;
+
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_trade_unread_page);
+
+    trade =
+        new Trade(
+            getIntent().getStringExtra("posterEmail"),
+            FIRESTORE_INSTANCE.document(getIntent().getStringExtra("posterItem")),
+            getIntent().getStringExtra("offeringEmail"),
+            FIRESTORE_INSTANCE.document(getIntent().getStringExtra("offeringItem")),
+            getIntent().getDoubleExtra("money", -1),
+            getIntent().getStringExtra("stateOfCompletion"));
+
+    getCurrentUser();
+    if (currentUser == null) {
+      goToLoginPage();
+    }
+    getCurrentUserInfo();
+    getXmlElements();
+
+    cancel_button.setOnClickListener(
```

Comment:
For consistency there should also be a back arrow image button on this page so the user knows they don't have to cancel

Reply:
Back arrow with finish() functionality added.